src/asprofPrivate.h

```diff
@@ -0,0 +1,19 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef _ASPROF_PRIVATE_H
+#define _ASPROF_PRIVATE_H
+
+#include "asprof.h"
+
+// Private functions for the native API.
+
+// Increment the thread-local sample counter, if it's already initialized.
+// If the counter is not already initialized, this function is a no-op.
+//
+// This function is signal-safe and safe to call racily with initialization.
+void asprofIncrementThreadLocalSampleCounter(void);
```

Comment:
With rare exceptions, non-local top-level functions are discouraged in async-profiler.
Static classes generally serve as namespaces. Actual C++ namespaces would be probably better, but that was a historical choice similar to the coding style adopted in HotSpot.